### DISCUSSION TITLE
6.2.x quickstart examples?

### DISCUSSION THREAD

**Discussion by bdlink** [resteasy/resteasy#3323] [Q&A] [answered]
When I was looking to upgrade my resteasy examples to EE10, I noticed they do not seem to be included in resteasy-6.2.1.Final-all.zip.  The resteasy-examples project has not been updated for 7 months and has a pull request https://github.com/resteasy/resteasy-examples/pull/92 saying it is being archived and directing the seeker to https://github.com/resteasy/resteasy-quickstarts.  Fair enough.  That is where the documents page directs one to find the examples.

However the resteasy-quickstarts page does not seem to have any examples, except for bootstrap with CDI (indeed the readme's list of examples is empty).  

Where does one look for the RESTEasy EE10 examples corresponding to the old examples?

Thanks

**Comment by jamezp** [comment, marked as answer]
Hi @bdlink, Sorry for the inconvenience there. I'm actually as we speak working on convert the resteasy-examples now.

The plan is going to be to archive/delete the resteasy-quickstarts and use continue using the resteasy-examples. On the initial commit there will not be much added to the examples. Really for Jakarta EE 10, there hasn't been anything added to Jakarta RESTful Web Services of interest. There are a few new utilities for stuff like the `EntityPart` API.

If there are specific examples you'd like to see please feel free to file an issue to add them and we can have a look at doing so.

**Comment by bdlink** [comment]
Thanks for your quick answer.  I have a unit in the fall where I teach RESTful services, and I usually use the current version of the RESTEasy examples.  I guess I can convert the ones I used last year to EE10 by hand.  I only use a few of the basic ones, from the oreilly-jaxrs-2-workbook so it should not be a big exercise (I had one of my examples using WF 27 Beta that I updated to EE10 with no issues).  

I have been dealing with upgrading my projects by creating an EE8 and EE10 branch.  This often results in having to change both branches when I make a substantive change, but I did not figure out a better way of handling this,  Would you go back to the examples project with a branch for EE10 or version 6 or whatever?

If I run into any issues, I will ask further.  

THanks,
Bruce

**Reply by jamezp** [reply]
I've definitely got a lot of experience making the transitions from javax to jakarta :) I've been maintaining two different branches really. That is what I do for WildFly Arqullian. Same with RESTEasy itself.

As for the examples. I actually moved the book examples to their own branch https://github.com/resteasy/resteasy-examples/tree/oreilly-jaxrs-2.0-workbook. The reason for that is the book was written to Jakarta REST 2.1 and I didn't want to ruin those examples.

I'm always happy to add more examples. So far I've converted the following locally:

- async-job-services
- bootable-jar
- examples-jsapi
- jaxb-json -> migrated to json-binding
- smime
- tracing-example

Currently these all run on WildFly. I'm going to add the example from https://github.com/resteasy/resteasy-quickstarts/tree/main/bootstrap-cdi too.

Would some more generic Jakarta EE examples be helpful? Having useful examples is my goal here so any suggestions are welcome :)

**Reply by bdlink** [reply]
I am currently converting a JSF-EJB-JPA example to use REST client -> REST services to publish restful endpoints on the session beans and consume these endpoints from the JSF web app.  The REST services is easy, as I have been doing it for years, but I am relatively new to the client API, so am progressing more slowly on the REST client side. (I know it is strange to use the REST client in a JSF web-application, but It is an illustration of decoupling the JSF side from the back end using REST instead of remote interfaces).

I found the tests in the examples (using the REST client) as useful as the code under test.   I am not wedded to the o'reilly books, but good books for up to date versions are not as easy to find as they once were.

I find examples that work are often as useful as documentation when starting.

**Reply by jamezp** [reply]
Okay excellent. It sounds like an example that uses multiple Jakarta EE API's would be useful. That sounds like a reasonable request to me.

I have submitted a [PR](https://github.com/resteasy/resteasy-examples/pull/96) which migrates some of the examples and deletes some. Once others have a look and that PR gets merged, we'll be adding more.

**Comment by bschatz** [comment]
Is there somewhere a simplest , minimalistic hello world example that works with a servlet-container (e.g. like tomcat) ?

Something like this -->
https://howtodoinjava.com/resteasy/resteasy-tomcat-hello-world-application/
would be fine, unfortunately it works not.

**Reply by liweinan** [reply]
@bschatz I have upgraded this example which you can use as reference:

- https://github.com/liweinan/resteasy-servlet30-demo

**Reply by jamezp** [reply]
Currently most of the examples either use WildFly, which is a full Jakarta EE container, or work with the new `SeBootstrap` API. However, we could create some more examples of how to work in other servlet containers.

**Reply by liweinan** [reply]
@jamezp Do you agree that I migrate the above example into our `resteasy-example` project?

**Reply by jamezp** [reply]
> @jamezp Do you agree that I migrate the above example into our `resteasy-example` project?

@liweinan I've not had a chance to look at it, but having a Tomcat example seems fine.

**Reply by liweinan** [reply]
@jamezp okay! I'll create a PR on this task for review :D

**Reply by bschatz** [reply]
My example is even a bit smaller, 
and it seems you need only the resteasy-servlet-initializer as dependency:

https://github.com/bschatz/resteasy-minimalistic-examples

This was the biggest hurdle for me, there were
so much instructions with needless and sometimes also wrong dependencies.

**Reply by liweinan** [reply]
@bschatz cool!

**Reply by liweinan** [reply]
I have created a PR here to add a minimal example: https://github.com/resteasy/resteasy-examples/pull/118
